extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -26,14 +28,88 @@ class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		$dependencies = array(
 			'lodash',
 			'wp-i18n',
 			'wp-token-list',
 		);
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+		$srcset_step         = 300;
+
+		if ( function_exists( 'jetpack_photon_url' )
+			&& preg_match_all( '/<img [^>]+>/', $content, $images )
+		) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					$orig_width  = $img_width[1];
+					$orig_height = $img_height[1];
+
+					// `src`s are already "photonized". Drop their parameters so we can add our own.
+					$src_parts = explode( '?', $img_src[1], 2 );
+					$orig_src  = $src_parts[0];
+
+					$srcset_parts = array();
+					if ( $is_squareish_layout ) {
+						$min_width = min( 600, $orig_width, $orig_height );
+						$max_width = min( 2000, $orig_width, $orig_height );
+
+						for ( $w = $min_width; $w <= $max_width; $w += $srcset_step ) {
+							$photonized_src = jetpack_photon_url(
+								$orig_src,
+								array(
+									'resize' => $w . ',' . $w,
+									'strip'  => 'all',
+								)
```

Comment:
Would WP [`add_query_arg`](https://developer.wordpress.org/reference/functions/add_query_arg/) be useful here? It can be used to replace existing URL query args.

Reply:
I think relying on the photon functions makes the most sense here. We're not directly manipulating query args, the function should handle that internally.

Reply:
I ended up using `add_query_args` because photon functions are for transforming non-photon URLs into photon URLs… which is weird.

`add_query_arg` allows manipulation of _existing photon URLs_, which is what we actually need 👍